test/helpers/kubectl.go

```diff
@@ -2207,8 +2207,10 @@ func (kub *Kubectl) RedeployKubernetesDnsIfNecessary(force bool) {
 	if err == nil && !force {
 		ginkgoext.By("Kubernetes DNS is up and operational")
 		return
-	} else {
+	} else if err != nil {
 		ginkgoext.By("Kubernetes DNS is not ready: %s", err)
+	} else {
+		ginkgoext.By("Kubernetes DNS is not ready (force=%v)", force)
```

Comment:
nit: I Think the non-nil `err` is what decides whether kube-dns is ready or not, so if the err is nil redeployment was requested then I think this would be more accurate:

```suggestion
		ginkgoext.By("Kubernetes DNS is ready, redeploying (force=%v)", force)
```